### PR TITLE
Authentication methods for any Auth Collection, not only Users

### PR DESCRIPTION
**Pull request by iluvadev**
Hi, 
First, apologies if I made more changes than strictly necessary. This is a proposal.

In PocketBase you can have multiple Auth Collections, and then Authenticate over them.
I created `CollectionAuthService` with authenticate methods with custom Collection name; `UserService` uses that class to authenticate to collection `users`

I have seen that the admin and user authentication methods only change in the url: I have abstracted it in `BaseAuthService`: `AdminService` and `CollectionAuthService` inherits from this class.
Then, `AuthModel` is a new model for Admin and User authentication responses

In short: it is possible to access the authentication methods of any Auth Collection
```csharp
client.AuthCollection("authors").AuthenticateWithPassword("my@email.com", "myPassword");
```

### CODE REVIEW

**Comment by iluvadev** [comment]
I close this PR: I'm working on that. 
I will try to generalize more the code in order to accept other Models for Auth Collections: with more fields for example

**Comment by PRCV1** [comment]
i am on mobile right now. am i missing something, why do i need this?
as long as i am authenticated, i set the headers for the authentication. why do i need to authenticate a specific collection?

**Comment by iluvadev** [comment]
In PocketBase you can authenticate as Admin or as User by default. But also, you can create an Auth Collection, for example "authors" and "clients" each Auth collection accepts authenticate over the collection. 
In the example, you could authenticate as an "author" or as a "client". This can be useful for collection rules.
Take a look: [auth-collection](https://pocketbase.io/docs/collections/#auth-collection)

**Comment by PRCV1** [comment]
ah i see, i guess this was added with the 0.8 update and i just completly overlooked this, haha.
thanks for implementing it :)
